src/Microsoft.Azure.WebJobs.Host/Queues/Listeners/QueueListener.cs

```diff
@@ -145,17 +146,36 @@ public void Dispose()
                 _stopWaitingTaskSource = new TaskCompletionSource<object>();
             }
 
-            IEnumerable<IStorageQueueMessage> batch;
+            IEnumerable<IStorageQueueMessage> batch  = null;
             try
             {
-                batch = await _queue.GetMessagesAsync(_queueProcessor.BatchSize,
-                    _visibilityTimeout,
-                    options: null,
-                    operationContext: null,
-                    cancellationToken: cancellationToken);
+                if (!_queueExists.HasValue || !_queueExists.Value)
```

Comment:
nit: consider using `if (!_queueExists.GetValueOrDefault(false))`

Reply:
I'll make that change